repo/blob/azure/azure_storage.go

```diff
@@ -274,9 +277,13 @@ func New(ctx context.Context, opt *Options, isCreate bool) (blob.Storage, error)
 		service, serviceErr = azblob.NewClientWithSharedKeyCredential(
 			fmt.Sprintf("https://%s/", storageHostname), cred, nil,
 		)
-
+	// Azure AD
```

Comment:
Should this should become an explicit option in the configuration, instead of being the default?

Reply:
The `DefaultAzureCredential` supports [several different credentials](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/environment_credential.go#L44-L68), how about that we only support some of them(e.g. only support `AZURE_TENANT_ID`, `AZURE_CLIENT_ID` and `AZURE_CLIENT_SECRET`) and let users provide them via the CLI options just like `SASToken` and `StorageKey`?

Reply:
@ywk253100 Yes, explicitly adding the needed support as you suggest seems to be the desire approach, at least at first. Let's go down that path. And please add tests for the new functionality.

As @leuyentran pointed out, this is adding the ability to use various authentication methods (often referred to as SDK default creds), not only Azure AD. The problem I see with default credentials is that it is not explicit, which is often harder to track when there are issues.

> DefaultAzureCredential is a default credential chain for applications that will deploy to Azure. It combines credentials suitable for deployment with credentials suitable for local development. It attempts to authenticate with each of these credential types, in the following order, stopping when one provides a token:
>
> [EnvironmentCredential](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#EnvironmentCredential)
> [WorkloadIdentityCredential](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#WorkloadIdentityCredential), if environment variable configuration is set by the Azure workload identity webhook. Use [WorkloadIdentityCredential](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#WorkloadIdentityCredential) directly when not using the webhook or needing more control over its configuration.
> [ManagedIdentityCredential](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#ManagedIdentityCredential)
> [AzureCLICredential](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#AzureCLICredential)

If we decide to allow the usage of "default" creds, then the comment above (`// Azure AD`) should be updated. Currently it is misleading.